src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -109,83 +110,100 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
             }
         }
 
-        if (request.getWorkflowId() == null) {
-            // Throttle incoming requests
-            checkMaxWorkflows(request.getRequestTimeout(), request.getMaxWorkflows(), ActionListener.wrap(max -> {
-                if (!max) {
-                    String errorMessage = "Maximum workflows limit reached " + request.getMaxWorkflows();
-                    logger.error(errorMessage);
-                    FlowFrameworkException ffe = new FlowFrameworkException(errorMessage, RestStatus.BAD_REQUEST);
-                    listener.onFailure(ffe);
-                    return;
-                } else {
-                    // Create new global context and state index entries
-                    flowFrameworkIndicesHandler.putTemplateToGlobalContext(templateWithUser, ActionListener.wrap(globalContextResponse -> {
-                        flowFrameworkIndicesHandler.putInitialStateToWorkflowState(
-                            globalContextResponse.getId(),
-                            user,
-                            ActionListener.wrap(stateResponse -> {
-                                logger.info("create state workflow doc");
-                                listener.onResponse(new WorkflowResponse(globalContextResponse.getId()));
+        flowFrameworkIndicesHandler.initGlobalContextIndexIfAbsent(ActionListener.wrap(indexCreated -> {
+            if (!indexCreated) {
+                listener.onFailure(new FlowFrameworkException("No response to create global_context index", INTERNAL_SERVER_ERROR));
+                return;
+            }
+            if (request.getWorkflowId() == null) {
+                // Throttle incoming requests
+                checkMaxWorkflows(request.getRequestTimeout(), request.getMaxWorkflows(), ActionListener.wrap(max -> {
+                    if (!max) {
+                        String errorMessage = "Maximum workflows limit reached " + request.getMaxWorkflows();
+                        logger.error(errorMessage);
+                        FlowFrameworkException ffe = new FlowFrameworkException(errorMessage, RestStatus.BAD_REQUEST);
+                        listener.onFailure(ffe);
+                        return;
+                    } else {
+                        // Create new global context and state index entries
+                        flowFrameworkIndicesHandler.putTemplateToGlobalContext(
```

Comment:
This method creates a GC index in it. `initGlobalContextIndexIfAbsent` method.

Reply:
yeah I originally thought I can just move it earlier and forgot to remove that init from putTemplateToGlobalContext but just took Josh's suggestion and added the check for if index exists earlier since no index means max workflows hasn't been reached